home/tests/test_whatsapp.py

```diff
@@ -53,3 +57,80 @@ def test_create_whatsapp_template_with_buttons(self):
 
         self.assertEquals(request.headers["Authorization"], "Bearer fake-access-token")
         self.assertEquals(request.body, json.dumps(data, indent=4))
+
+    @responses.activate
+    def test_create_whatsapp_template_with_image(self):
+        img_name = "test.jpeg"
+        img_path = f"{settings.TEST_STATIC_ROOT}/{img_name}"
+
+        f = open(img_path, "rb")
+        image = Image(
+            title=img_name,
+            file=ImageFile(f, name=img_path),
+        )
+        mock_file_content = image.file
+
+        image.save()
+        f.seek(0)
+
+        mock_session_id = "TEST_SESSION_ID"
+        mock_session_url = "http://whatsapp/graph/v14.0/app/uploads"
+        responses.add(responses.POST, mock_session_url, json={"id": mock_session_id})
+
+        mock_start_upload_url = f"http://whatsapp/graph/{mock_session_id}"
+        mock_image_handle = "TEST_IMAGE_HANDLE"
+        req_files = {
+            "file": mock_file_content,
+            "number": settings.FB_BUSINESS_ID,
+            "access_token": settings.WHATSAPP_ACCESS_TOKEN,
+        }
+        responses.add(
+            responses.POST,
+            mock_start_upload_url,
+            json={"h": mock_image_handle},
+            match=[
+                multipart_matcher(
+                    req_files,
+                )
+            ],
+        )
+
+        template_url = "http://whatsapp/graph/v14.0/27121231234/message_templates"
+        responses.add(responses.POST, template_url, json={})
+        #  To run the WhatsAppTests on their own, the ID of 2 below should be set to 1
+        #  This is due to other content being inported if you run the full set of tests
+        create_whatsapp_template("Test-Template", "Test Body", [], 2)
```

Comment:
Can we not use `image.id` here, instead of the hardcoded `2`? And then get rid of the comment

Reply:
:facepalm: Yes, tired brain wasn't thinking straight :joy: